src/super_gradients/training/transforms/detection/abstract_detection_transform.py

```diff
@@ -0,0 +1,36 @@
+import abc
+from abc import abstractmethod
+from typing import List
+
+from super_gradients.training.samples import DetectionSample
+
+__all__ = ["AbstractDetectionTransform"]
+
+
+class AbstractDetectionTransform(abc.ABC):
+    """
+    Base class for all transforms for object detection sample augmentation.
+    """
+
+    def __init__(self, additional_samples_count: int = 0):
+        """
+        :param additional_samples_count: (int) number of samples that must be extra samples from dataset. Default value is 0.
+        """
+        self.additional_samples_count = additional_samples_count
+
+    @abstractmethod
+    def apply_to_sample(self, sample: DetectionSample) -> DetectionSample:
```

Comment:
why are we calling apply_to_sample instead of call ?
For backward compatibility (which, is not at such a high risk IMO since theese are only been called from within DetectionDataset..), maybe its better to use __call__() with some processing in the beginning according to instance of sample.